libyul/backends/wasm/WasmCodeTransform.cpp

```diff
@@ -253,6 +253,7 @@ wasm::Expression WasmCodeTransform::operator()(ForLoop const& _for)
 	m_breakContinueLabelNames.push({breakLabel, continueLabel});
 
 	yul::Type conditionType = m_typeInfo.typeOf(*_for.condition);
+	yulAssert(conditionType == "i32"_yulstring || conditionType == "i64"_yulstring, "");
```

Comment:
Probably can be removed.

Reply:
Yeah. I had something like this in my PR initially but @chriseth suggested to just check if the whole instruction exists. So I replaced it with:

``` c++
YulString eqz_instruction = YulString(conditionType.str() + ".eqz");
yulAssert(WasmDialect::instance().builtin(eqz_instruction), "");
```